### PR TITLE
Hotplug the cpus in random order by selecting random core

### PR DESCRIPTION
**Pull request by shriyak**
The script will select random core from the list, picks the random cpus and does
offline and online in random order for that core.Make sure that all cores
are online.Script will set smt to maximum value,later it
works in random order to offline and online the cpu for 100 iterations.

Signed-off-by: Shriya <shriyak@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by shriyak** [comment]
Old PR : https://github.com/avocado-framework-tests/avocado-misc-tests/pull/613

**Comment by PraveenPenguin** [comment]
@abdhaleegit can you please review this

File: cpu/em_cpuhotplug.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+                                           "| awk '{print $4}'", shell=True)
+        self.SOCKETS = process.system_output("lscpu | grep 'Socket(s):'"
+                                             "| awk '{print $2}'", shell=True)
+        self.THREADS = process.system_output("lscpu | grep 'Thread(s) per core"
+                                             ":'| awk '{print $4}'",
+                                             shell=True)
+        self.T_CORES = int(self.CORES) * int(self.SOCKETS)
+        self.log.info(" Cores = %s and threads = %s "
+                      % (self.T_CORES, self.THREADS))
+        process.system("echo 8 > /proc/sys/kernel/printk", shell=True)
```
Comment (PraveenPenguin) [inline cpu/em_cpuhotplug.py]: please add `ignore_status=True`

File: cpu/em_cpuhotplug.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+    def test(self):
+        """
+        This script picks a random core and then offlines all its threads
+        in a random order and onlines all its threads in a random order.
+        """
+        for x in range(1, 100):
+            self.log.info("================= TEST %s ==================" % x)
+            core_list = []
+            core_list = self.random_gen_cores()
+            for core in core_list:
```
Comment (PraveenPenguin) [inline cpu/em_cpuhotplug.py]: can it be 
`for core in self.random_gen_cores()`

File: cpu/em_cpuhotplug.py [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+        nums = [x for x in range(self.max_smt * core,
+                ((self.max_smt * core) + self.max_smt))]
+        random.shuffle(nums)
+        return nums
+
+    def offline_cpu(self, cpu_num):
+        """
+        Offline the particular cpu
+        """
+        cpu.offline(cpu_num)
```
Comment (PraveenPenguin) [inline cpu/em_cpuhotplug.py]: i think it better to check status of output of this command

File: cpu/em_cpuhotplug.py (hunk 2 above)
Comment (abdhaleegit) [reply cpu/em_cpuhotplug.py]: What you need is a list of shuffled random cpus between 1 to 100 ?
core_list = [ 2, 99, 34, 53, 56, 22, 12. ......]

the self.random_gen_cores() returns a list so no need to initialize core_list = []

File: cpu/em_cpuhotplug.py (hunk 3 above)
Comment (abdhaleegit) [reply cpu/em_cpuhotplug.py]: cpu.offline already uses _get_cpu_status() to check status and return 0 else 1

File: cpu/em_cpuhotplug.py [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+            self.log.info("Failed to offline the cpu %s" % cpu_num)
+        else:
+            self.log.info("Offline the cpu : %s" % cpu_num)
+
+    def online_cpu(self, cpu_num):
+        """
+        Online the particular cpu
+        """
+        cpu.online(cpu_num)
+        if cpu._get_cpu_status(cpu_num) is False:
```
Comment (abdhaleegit) [inline cpu/em_cpuhotplug.py]: this will be like this:
if not cpu.online(cpu_num):
    self.nfail += 1

File: cpu/em_cpuhotplug.py [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+                ((self.max_smt * core) + self.max_smt))]
+        random.shuffle(nums)
+        return nums
+
+    def offline_cpu(self, cpu_num):
+        """
+        Offline the particular cpu
+        """
+        cpu.offline(cpu_num)
+        if cpu._get_cpu_status(cpu_num) is True:
```
Comment (abdhaleegit) [inline cpu/em_cpuhotplug.py]: if cpu.offline(cpu_num):

File: cpu/em_cpuhotplug.py (hunk 2 above)
Comment (shriyak) [reply cpu/em_cpuhotplug.py]: Yes, @abdhaleegit @PraveenPenguin  : I will make this change

**Comment by PraveenPenguin** [comment]
@abdhaleegit thanks for review

**Pull request merged**